### ISSUE TITLE
[Week 9] v1.2: Somethings you can consider

### ISSUE THREAD

**Issue by okkhoy** [nusCS2113-AY1819S1/forum#54]
## TL/DR version:
- If Travis builds are failing, **for now**, it is OK to remove some of the tests
- You can add them back later 

## Longer version:
- One of the requirements we have for each milestone is that the build should be passing
- We understand that there may be a huge code churn at this moment as you are evolving the product from AB3/4 to your own (either enhancement or morphing)
- As a result, many of your tests may be failing.
- **For now**, it is OK to remove such failing tests. You can add them back later. <br><br>
- Few things to consider while evolving your product:
  - Update your tests as and when you refactor code from AB3/4
  - Commit frequently (as frequent as after every refactoring is good!)
  - Run the tests and see that the product is not broken due to refactoring<br><br>
- If you have already done a lot of refactoring:
  - Checkout the individual commits where you did refactoring
  - Check if the tests pass
  - Use branching to fix the tests and proceed to merge incrementally

## By when should the tests be brought back?
- Ensure to have some tests, which each of the team member has written, merged to your product master branch by v1.3
- **v1.3** will be used for the first round of testing (more on that in the lecture this week!)
